### ISSUE TITLE
Matplotlib version: AttributeError: module 'sip' has no attribute 'setapi' 

### ISSUE THREAD

**Issue by talhaanwarch** [awslabs/autogluon#1076]
I am not sure whether this error should be fixed in autogluon or not. But I am mentioning it as it throws an error but did not halt the code progress. This has been observed both in `v0.1` and `v0.2` of autogluon.
OS: Windows 10
Python: 3.8.5

Error log
```
Fitting model: NeuralNetFastAI ... Training model for up to 18.02s of the 18.01s of remaining time.
        Warning: Exception caused NeuralNetFastAI to fail during training... Skipping this model.
                module 'sip' has no attribute 'setapi'
Detailed Traceback:
Traceback (most recent call last):
  File "D:\tutorials\automl\venv1\lib\site-packages\autogluon\tabular\trainer\abstract_trainer.py", line 914, in _train_and_save
    model = self._train_single(X, y, model, X_val, y_val, **model_fit_kwargs)
  File "D:\tutorials\automl\venv1\lib\site-packages\autogluon\tabular\trainer\abstract_trainer.py", line 886, in _train_single
    model.fit(X=X, y=y, X_val=X_val, y_val=y_val, **model_fit_kwargs)
  File "D:\tutorials\automl\venv1\lib\site-packages\autogluon\core\models\abstract\abstract_model.py", line 411, in fit
    self._fit(**kwargs)
  File "D:\tutorials\automl\venv1\lib\site-packages\autogluon\tabular\models\fastainn\tabular_nn_fastai.py", line 164, in _fit
    try_import_fastai()
  File "D:\tutorials\automl\venv1\lib\site-packages\autogluon\core\utils\try_import.py", line 106, in try_import_fastai
    import autogluon.tabular.models.fastainn.imports_helper
  File "D:\tutorials\automl\venv1\lib\site-packages\autogluon\tabular\models\fastainn\imports_helper.py", line 1, in <module>
    from fastai.tabular.all import *
  File "D:\tutorials\automl\venv1\lib\site-packages\fastai\tabular\all.py", line 1, in <module>
    from ..basics import *
  File "D:\tutorials\automl\venv1\lib\site-packages\fastai\basics.py", line 1, in <module>
    from .data.all import *
  File "D:\tutorials\automl\venv1\lib\site-packages\fastai\data\all.py", line 1, in <module>
    from ..torch_basics import *
  File "D:\tutorials\automl\venv1\lib\site-packages\fastai\torch_basics.py", line 9, in <module>
    from .imports import *
  File "D:\tutorials\automl\venv1\lib\site-packages\fastai\imports.py", line 25, in <module>
    import requests,yaml,matplotlib.pyplot as plt,pandas as pd,scipy
  File "D:\tutorials\automl\venv1\lib\site-packages\matplotlib\pyplot.py", line 2469, in <module>
    switch_backend(rcParams["backend"])
  File "D:\tutorials\automl\venv1\lib\site-packages\matplotlib\__init__.py", line 619, in __getitem__
    plt.switch_backend(rcsetup._auto_backend_sentinel)
  File "D:\tutorials\automl\venv1\lib\site-packages\matplotlib\pyplot.py", line 257, in switch_backend
    switch_backend(candidate)
  File "D:\tutorials\automl\venv1\lib\site-packages\matplotlib\pyplot.py", line 277, in switch_backend
    class backend_mod(matplotlib.backend_bases._Backend):
  File "D:\tutorials\automl\venv1\lib\site-packages\matplotlib\pyplot.py", line 278, in backend_mod
    locals().update(vars(importlib.import_module(backend_name)))
  File "C:\Anaconda3\lib\importlib\__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "D:\tutorials\automl\venv1\lib\site-packages\matplotlib\backends\backend_qt5agg.py", line 11, in <module>
    from .backend_qt5 import (
  File "D:\tutorials\automl\venv1\lib\site-packages\matplotlib\backends\backend_qt5.py", line 13, in <module>
    import matplotlib.backends.qt_editor.figureoptions as figureoptions
  File "D:\tutorials\automl\venv1\lib\site-packages\matplotlib\backends\qt_editor\figureoptions.py", line 11, in <module>
    from matplotlib.backends.qt_compat import QtGui
  File "D:\tutorials\automl\venv1\lib\site-packages\matplotlib\backends\qt_compat.py", line 174, in <module>
    _setup()
  File "D:\tutorials\automl\venv1\lib\site-packages\matplotlib\backends\qt_compat.py", line 134, in _setup_pyqt4
    _setup_pyqt4_internal(api=2)
  File "D:\tutorials\automl\venv1\lib\site-packages\matplotlib\backends\qt_compat.py", line 117, in _setup_pyqt4_internal
    sip.setapi(_sip_api, api)
AttributeError: module 'sip' has no attribute 'setapi'
```
[This StackOverflow issue](https://stackoverflow.com/questions/64291087/matplotlib-module-sip-has-no-attribute-setapi) suggests downgrading matplotlib version to `3.2` to resolve this issue. 

Hardcoding the matplotlib version in [setup.py](https://github.com/awslabs/autogluon/blob/6daf6d464db03cd77412a001c5f99b91eee9042e/core/setup.py#L32) may fix the issue.

**Comment by Innixma** [comment]
Could you try to get a working version by manually downgrading matplotlib after installing AutoGluon?

**Comment by talhaanwarch** [comment]
First, I uninstalled matplotlib version `3.4.1`, then install `3.2.0`, this resolved the issue

**Comment by Innixma** [comment]
Thanks for the info!

I'm leaning towards not fixing this in AutoGluon as it is Windows specific and seems like a bug in FastAI / matplotlib that should be fixed on their end. If it remains a problem long-term, we may consider restricting the version of matplotlib.

As of right now, you may want to open an issue with FastAI or matplotlib so they have visibility to the problem.

**Comment by talhaanwarch** [comment]
Thanks. I opened the issue with [matplotlib ](https://github.com/matplotlib/matplotlib/issues/20040) suggesting that simply installing pyqt5 will resolve this error. So here at autogluon, i suggest adding `pyqt5` in the requirement file or setup file.

**Comment by Innixma** [comment]
Thanks! I'll wait to see matplotlib's response before adding additional dependencies.

**Comment by Innixma** [comment]
Appears that this is fixed in https://github.com/matplotlib/matplotlib/pull/20096

@talhaanwarch, I'm closing this for now. Please re-open if the issue is not addressed in master branch AutoGluon / v0.3
